ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1052,187 @@ def test_server_option_with_unreachable_ad(self):
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+    num_ad_subdomains = 0
+    num_ad_treedomains = 0
+    uid_override = "99999999"
+    gid_override = "78878787"
+
+    def test_add_nonposix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_groups_default_trusted_range(self, type):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true/hybrid/false and test
+        user with no posix attributes.
+        """
+        self.mod_idrange_auto_private_group(type)
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            test_group = self.clients[0].run_command(["id", nonposixuser])
+            gid_str = "gid={0}(domain users@{1})".format(gid, self.ad_domain)
+            grp_str = "groups={0}(domain users@{1})".format(gid,
+                                                            self.ad_domain)
+            assert gid_str in test_group.stdout_text
+            assert grp_str in test_group.stdout_text
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        with self.set_idoverrideuser(nonposixuser,
+                                     self.uid_override,
+                                     self.gid_override
+                                     ):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.clients[0], nonposixuser)
+            assert (uid == self.uid_override and gid == self.gid_override)
+            test_group = self.clients[0].run_command(
+                ["id", nonposixuser]).stdout_text
+            if type == "hybrid":
+                version = tasks.get_sssd_version(self.clients[0])
+                with xfail_context(version <= tasks.parse_version('2.6.3'),
+                                   'https://github.com/SSSD/sssd/issues/5989'):
+                    assert "domain users@{0}".format(
+                        self.ad_domain) in test_group.stdout_text
```

Comment:
test_group is already obtained from run_command(...).stdout_text, this line should rather be:
```
                    assert "domain users@{0}".format(
                        self.ad_domain) in test_group
```

The logic can also be simplified: no need to do `if type == "hybrid"`, you can use instead:
```
            version = tasks.get_sssd_version(self.clients[0])
            with xfail_context(version <= tasks.parse_version('2.6.3') and type == "hybrid",
                               'https://github.com/SSSD/sssd/issues/5989'):
                assert "domain users@{0}".format(self.ad_domain) in test_group
```